packages/fiori/src/FlexibleColumnLayout.ts

```diff
@@ -0,0 +1,821 @@
+	* @sap-restricted
+	*/
+	@property({ type: Object, defaultValue: undefined })
+	_layoutsConfiguration?: LayoutConfiguration;
+
+	/**
+	* Defines the content in the start column.
+	* @type {HTMLElement}
+	* @slot
+	* @name sap.ui.webc.fiori.FlexibleColumnLayout.prototype.startColumn
+	* @public
+	*/
+	@slot()
+	startColumn!: Array<HTMLElement>;
+
+	/**
+	* Defines the content in the middle column.
+	* @type {HTMLElement}
+	* @slot
+	* @name sap.ui.webc.fiori.FlexibleColumnLayout.prototype.midColumn
+	* @public
+	*/
+	@slot()
+	midColumn!: Array<HTMLElement>;
+
+	/**
+	* Defines the content in the end column.
+	* @type {HTMLElement}
+	* @slot
+	* @name sap.ui.webc.fiori.FlexibleColumnLayout.prototype.endColumn
+	* @public
+	*/
+	@slot()
+	endColumn!: Array<HTMLElement>;
+
+	initialRendering: boolean;
+	_handleResize: () => void;
+	static i18nBundle: I18nBundle;
+	_prevLayout: FCLLayout | null;
+
+	constructor() {
+		super();
+
+		this._prevLayout = null;
+		this.initialRendering = true;
+		this._handleResize = this.handleResize.bind(this);
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return FlexibleColumnLayoutCss;
+	}
+
+	static get template() {
+		return FlexibleColumnLayoutTemplate;
+	}
+
+	static get dependencies() {
+		return [Button];
+	}
+
+	static async onDefine() {
+		FlexibleColumnLayout.i18nBundle = await getI18nBundle("@ui5/webcomponents-fiori");
+	}
+
+	static get ANIMATION_DURATION() {
+		return getAnimationMode() !== AnimationMode.None ? 560 : 0;
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._handleResize.bind(this));
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._handleResize);
+	}
+
+	onAfterRendering() {
+		if (this.initialRendering) {
+			this.handleInitialRendering();
+			return;
+		}
+
+		this.syncLayout();
+	}
+
+	handleInitialRendering() {
+		this._prevLayout = this.layout;
+		this.updateLayout();
+		this.initialRendering = false;
+	}
+
+	handleResize() {
+		if (this.initialRendering) {
+			return;
+		}
+
+		// store the previous layout
+		const prevLayoutHash = this.columnLayout!.join();
+
+		// update the column layout, based on the current width
+		this.updateLayout();
+
+		// fire layout-change if the column layout changed
+		if (prevLayoutHash !== this.columnLayout!.join()) {
+			this.fireLayoutChange(false, true);
+		}
+	}
+
+	startArrowClick() {
+		this.arrowClick({ start: true, end: false });
+	}
+
+	endArrowClick() {
+		this.arrowClick({ start: false, end: true });
+	}
+
+	arrowClick(options: { start: boolean, end: boolean }) {
+		// update public property
+		this.layout = this.nextLayout(this.layout, { start: options.start, end: options.end })!;
+
+		// update layout
+		this.updateLayout();
+
+		// fire layout-change
+		this.fireLayoutChange(true, false);
+	}
+
+	updateLayout() {
+		this._width = this.widthDOM;
+		this._columnLayout = this.nextColumnLayout(this.layout);
+		this._visibleColumns = this.calcVisibleColumns(this._columnLayout);
```

Comment:
As discussed offline, calcVisibleColumns will now return 3 for 
["10%", "90%", 0], while previously 2,  as it internally checks  for "0px" not 0 as before. This will lead to wrong value for the _visibleColumns internal property and attribute and wrong styles applied